addons/web_editor/static/tests/link_test.js

```diff
@@ -0,0 +1,458 @@
+import { patchWithCleanup, nextTick, triggerHotkey } from "@web/../tests/helpers/utils";
+import { Wysiwyg } from "@web_editor/js/wysiwyg/wysiwyg";
+import {
+    triggerEvent,
+    insertText,
+    insertParagraphBreak,
+} from "@web_editor/js/editor/odoo-editor/test/utils";
+
+function onMount() {
+    const editor = wysiwyg.odooEditor;
+    const editable = editor.editable;
+    editor.testMode = true;
+    return { editor, editable };
+}
+
+function inputText(selector, content, { replace = false } = {}) {
+    if (replace) {
+        document.querySelector(selector).value = "";
+    }
+    document.querySelector(selector).focus();
+    for (const char of content) {
+        document.execCommand("insertText", false, char);
+        document
+            .querySelector(selector)
+            .dispatchEvent(new window.KeyboardEvent("keydown", { key: char }));
+        document
+            .querySelector(selector)
+            .dispatchEvent(new window.KeyboardEvent("keyup", { key: char }));
+    }
+}
+
+let serverData;
+let wysiwyg;
+
+QUnit.module(
+    "web_editor",
+    {
+        before: function () {
+            serverData = {
+                models: {
+                    note: {
+                        fields: {
+                            body: {
+                                string: "Editor",
+                                type: "html",
+                            },
+                        },
+                        records: [
+                            {
+                                id: 1,
+                                display_name: "first record",
+                                body: "<p><br></p>",
+                            },
+                        ],
+                    },
+                },
+            };
+        },
+        beforeEach: async function () {
+            setupViewRegistries();
+            patchWithCleanup(Wysiwyg.prototype, {
+                init() {
+                    super.init(...arguments);
+                    wysiwyg = this;
+                },
+            });
+            await makeView({
+                type: "form",
+                serverData,
+                resModel: "note",
+                arch:
+                    "<form>" +
+                    '<field name="body" widget="html" style="height: 100px"/>' +
+                    "</form>",
+                resId: 1,
+            });
+        },
+    },
+    function () {
+        QUnit.module("HotKeys");
+
+        QUnit.test("should be able to create link with ctrl+k and ctrl+k", async function (assert) {
+            const { editor, editable } = onMount();
+            const node = editable.querySelector("p");
+            setSelection(node, 0);
+            await triggerHotkey("control+k");
+            inputText(".o_command_palette_search", "k");
+            await nextTick();
+            triggerEvent(node, "keydown", {
+                key: "Enter",
+            });
+            await nextTick();
```

Comment:
I don't think we need to test this flow, the one below should be enough.
But if you still would like to keep it, then I suggest you input "Create link" as text, instead of just "k", as it would make it more self-documenting.
Now triggering the event on `node`, being `node` a P in the editable... I guess that works, but it's weird, considering we have a modal in front of it at this point (in a real scenario the keydown event would not have been triggered on that P). Try to do `await triggerHotkey("Enter")` instead, and you should not need to await nextTick after it!